File: CSharp/Patterns/RingPattern.cs
using System;

namespace CSharp
{
    public class RingPattern : RTPattern
    {
        public RingPattern(Color a, Color b) : base(a, b)
        {
            // Nothing to do
        }

        public override Color PatternAt(Point p)
        {
            return (int) Math.Floor(Math.Sqrt(p.x * p.x + p.z * p.z)) % 2 == 0 ? a : b;
        }

        public override Color UVPatternAt(double u, double v, CubeFace face = CubeFace.None)
        {
            return Color.Black;
        }
    }
}

File: Java/src/main/java/Patterns/RingPattern.java
package RTC.Patterns;

import RTC.Math.*;
import RTC.Material.*;

public class RingPattern extends Pattern {
    private Color a, b;

    public RingPattern(Color A, Color B) {
        super();

        a = A;
        b = B;
    }

    public Color patternAt(Point point) {
        double value = Math.sqrt(point.getX() * point.getX() + point.getZ() * point.getZ());
        if (Math.floor(value) % 2 == 0) {
            return a;
        } else {
            return b;
        }
    }
}